example/viam_example_app/lib/main.dart

```diff
@@ -102,22 +102,34 @@ class _MyHomePageState extends State<MyHomePage> {
     });
   }
 
-  //TODO refactor to make all _is methods the same
-
-  bool _isCamera(ResourceName rn) {
-    return rn.subtype == 'camera';
-  }
-
-  bool _isSensor(ResourceName rn) {
-    return rn.subtype == Sensor.subtype.resourceSubtype || rn.subtype == MovementSensor.subtype.resourceSubtype;
+  StreamClient _getStream(ResourceName name) {
+    return _robot.getStream(name.name);
   }
 
-  bool _isServo(ResourceName rn) {
-    return rn.subtype == Servo.subtype.resourceSubtype;
+  bool _isImplemented(ResourceName rname) {
+    return [
+      Camera.subtype.resourceSubtype,
+      MovementSensor.subtype.resourceSubtype,
+      Sensor.subtype.resourceSubtype,
+      Servo.subtype.resourceSubtype,
+    ].contains(rname.subtype);
   }
 
-  StreamClient _getStream(ResourceName name) {
-    return _robot.getStream(name.name);
+  Widget? _getScreen(ResourceName rname) {
```

Comment:
its a simple example app, so maybe too big of a undertaking than what its worth, but perhaps making some named routes would be an elegant way to handle this and scale as we get a few more examples. Not too worried about it for now though.

Reply:
I want to have this app be as simple as possible, so that anyone building a fresh flutter app can very easily follow along. I"m not sure how much complication routing adds, so it might be worth looking into. But this isn't supposed to be a *good* app, it's just supposed to show how to use our APIs. The *good* app comes later